reporters/ReportVectorMigration.cpp

```diff
@@ -166,33 +227,60 @@ GetReportInstantiator( Kernel::instantiator_function_t* pif )
             throw QueryInterfaceException(__FILE__, __LINE__, __FUNCTION__, "pvc", "IMigrate", "IVectorCohort");
         }
 
+        VectorStateEnum::Enum state = pvc->GetState();
+        if (!m_MustBeInState.empty())
+        {
+            if (!std::count(m_MustBeInState.begin(), m_MustBeInState.end(), state)) 
+            {
+                // state is not one of the required states
+                return;
+            }
+        }
+
         uint32_t vci_id = pvc->GetID();
         int from_node_id = pSim->GetNodeExternalID( nodeSuid ) ;
+        if (!m_MustBeFromNode.empty())
+        {
+            if (!std::count(m_MustBeFromNode.begin(), m_MustBeFromNode.end(), from_node_id))
+            {
+                // from_node_id is not one we want in the report
+                return;
+            }
+        }
         int to_node_id = pSim->GetNodeExternalID( pim->GetMigrationDestination() ) ;
-        int mig_type = pim->GetMigrationType() ;
-        VectorStateEnum::Enum state = pvc->GetState();
+        if (!m_MustBeToNode.empty())
+        {
+            if (!(std::find(m_MustBeToNode.begin(), m_MustBeToNode.end(), to_node_id) != m_MustBeToNode.end()))
```

Comment:
why did you switch from count to find?

I'm really wondering if there is a way to create a function that does this test.

Reply:
I will look at std::count vs find. i think this one is find because of your previous (on dtktrunk) comment. I'll standardize them, though.

Reply:
moved everything to find without NOT